modules/integration/tests-common/admin-clients/src/main/java/org/wso2/bps/integration/common/clients/bpmn/ActivitiRestClient.java

```diff
@@ -167,16 +146,10 @@ public void waitForTaskGeneration() throws InterruptedException {
      * @return String with the Status
      * @throws IOException
      */
-
     public String unDeployBPMNPackage(String deploymentID) throws IOException {
-
         String url = serviceURL + "repository/deployments/"
                      + deploymentID;
-        HttpHost target = new HttpHost(hostname, port, "http");
-        DefaultHttpClient httpClient = new DefaultHttpClient();
-        httpClient.getCredentialsProvider().setCredentials
-                (new AuthScope(target.getHostName(), target.getPort()),
-                 new UsernamePasswordCredentials(username, password));
+        DefaultHttpClient httpClient = getHttpClient();
```

Comment:
Instead of creating httpClient objects per invocation, a better idea would be to use a single instance initialilzed with a "MultiThreadedConnectionManager". Maybe you can use some configuration file too to make some parameters such as max Http connections allowed, etc configurable.